File: BKabanApi/BKabanApi/DB/UserDataRepository.cs
using System.Data;
using Dapper;
using Microsoft.Data.SqlClient;

namespace BKabanApi.Models
{
    public interface IUserDataRepository
    {
        UserDataModel GetUserData(int userId);
    }

    public class UserDataRepository : IUserDataRepository
    {
        private readonly string _connectionString;

        public UserDataRepository(string connectionString)
        {
            _connectionString = connectionString;
        }

        public UserDataModel GetUserData(int userId)
        {
            using IDbConnection db = new SqlConnection(_connectionString);
            var userData = new UserDataModel
            {
                Boards = db.Query<BoardModel>(
                    @"SELECT B.Id, B.Name
                    FROM BoardTable AS B
                    JOIN UserTable AS U ON U.Id = B.UserId
                    WHERE U.Id = @userId
                    ORDER BY Position", new {userId})
            };

            return userData;
        }
    }
}

File: BKabanApiJava/src/main/java/edu/bkaban/repositories/UserDataRepository.java
package edu.bkaban.repositories;

import edu.bkaban.models.UserDataModel;
import edu.bkaban.models.board.BoardModel;
import edu.bkaban.services.DbService;
import org.springframework.stereotype.Service;

import java.sql.SQLException;
import java.util.LinkedList;

@Service
public class UserDataRepository {
    private final String getUserDataSql =
            "SELECT B.Id, B.Name\n" +
                    "FROM BoardTable AS B\n" +
                    "JOIN UserTable AS U ON U.Id = B.UserId\n" +
                    "WHERE U.Id = ?\n" +
                    "ORDER BY Position";

    public UserDataModel getUserData(int userId) throws SQLException {
        try (var conn = DbService.getConnection(); var stmt = conn.prepareStatement(getUserDataSql)) {
            stmt.setInt(1, userId);

            var userData = new UserDataModel();
            userData.setBoards(new LinkedList<BoardModel>());

            var rset = stmt.executeQuery();
            while (rset.next()) {
                var board = new BoardModel();
                board.setId(rset.getInt("Id"));
                board.setName(rset.getString("Name"));
                userData.getBoards().add(board);
            }

            return userData;
        }
    }
}
